### ISSUE TITLE
bug: [Windows] Cannot clean logs files or open after started server / stopped server

### ISSUE THREAD

**Issue by louis-jan** [janhq/jan#1904]
**Describe the bug**
I can't open, clean or delete the logs without closing the app after started server. 

**Steps to reproduce**
1. Go to API Server
2. Click on Start Server
3. Stop Server
4. Go to settings
5. Clean logs
6. No message toast for the action

**Expected behavior**

**Screenshots**
![image](https://github.com/janhq/jan/assets/133622055/bff32018-4609-4f05-9380-26e3595b4e3d)

**Environment details**
- Operating System: [Specify your OS. e.g., MacOS Sonoma 14.2.1, Windows 11, Ubuntu 22, etc]
- Jan Version: [e.g., 0.4.3]
- Processor: [e.g., Apple M1, Intel Core i7, AMD Ryzen 5, etc]
- RAM: [e.g., 8GB, 16GB]
- Any additional relevant hardware specifics: [e.g., Graphics card, SSD/HDD]

**Logs**
If the cause of the error is not clear, kindly provide your usage logs:
- `tail -n 50 ~/jan/logs/app.log` if you are using the UI
- `tail -n 50 ~/jan/logs/server.log` if you are using the local api server
Making sure to redact any private information.

**Additional context**
Add any other context or information that could be helpful in diagnosing the problem.

**Comment by 0xgokuz** [comment]
I'm going to take a look on this 😁

**Comment by 0xgokuz** [comment]
Would like to add more clarification after quick discussion with @louis-jan 

### Problems
- Clean Log button does not Delete `server.log`

### Expected
- Clean Log button should Delete the `server.log`
- Starting the Server should Create, and keep writting to the `server.log`

**Comment by 0xgokuz** [comment]
I'm on `Mac OS Sonoma 14.2.1 (23C71)`, seems like it works with me. So I'm guessing this is Windows specific looking at your screenshot @louis-jan 

This is after I started the server
```
~/jan on ☁️  (ap-southeast-1)
❯ ls -l logs/server.log
-rw-r--r--  1 0xgokuz  staff  7374 Feb 11 07:13 logs/server.log
```

This is when I stopped, and then executed the **Clear** button
```
~/jan on ☁️  (ap-southeast-1)
❯ ls -l logs/server.log
-rw-r--r--  1 0xgokuz  staff  0 Feb 11 07:13 logs/server.log
```

**Comment by louis-jan** [comment]
> I'm on `Mac OS Sonoma 14.2.1 (23C71)`, seems like it works with me. So I'm guessing this is Windows specific looking at your screenshot @louis-jan
> 
> This is after I started the server
> 
> ```
> ~/jan on ☁️  (ap-southeast-1)
> ❯ ls -l logs/server.log
> -rw-r--r--  1 0xgokuz  staff  7374 Feb 11 07:13 logs/server.log
> ```
> 
> This is when I stopped, and then executed the **Clear** button
> 
> ```
> ~/jan on ☁️  (ap-southeast-1)
> ❯ ls -l logs/server.log
> -rw-r--r--  1 0xgokuz  staff  0 Feb 11 07:13 logs/server.log
> ```

Forgot to mention, this issue only happens on Windows

**Comment by louis-jan** [comment]
Can't reproduce after the state loop fix #2017
